### ISSUE TITLE
NDepend Code Quality Rules

### ISSUE THREAD

**Issue by Niols** [Evernest/Evernest#35]
I'm listing here the errors found in the NDepend report.
More details in `doc/Cours/Présentations/NDepend/NDependOut/NDependReport.html`
- [ ] Avoid namespaces mutually dependent
  - [ ] EvernestAPI shouldn't use EvernestAPI.Providers
  - [x] EvernestFront.Exceptions shoudn't use EvernestFront
- [ ] Potentially dead Types
  - [x] EvernestBack.AzureStorageClient
  - [x] EvernestBack.Test
  - [x] EvernestBack.EventStream
  - [x] EvernestBack.Reader
- [ ] Potentially dead Methods
  Too many things to say here, essentially in:
  - [ ] EvernestFront
  - [ ] EvernestBack
- [x] Constructors of abstract classes should be declared as protected or private
  - [x] EvernestBack.Agent
- [x] Avoid having different types with same name
  - [x] EvernestFront.Stream

These are only the critical rules. Again, take a look at `doc/Cours/Présentations/NDepend/NDependOut/NDependReport.html`.

**Comment by eliemichel** [comment]
@Niols Do you regularely update NDependReport.html?

**Comment by Niols** [comment]
@eliemichel No, because I can't. I only have a 14-days trial. You can download your trial here: http://www.ndepend.com/download. If you want me to do this on your computer, I can
